osu.Game/Skinning/Components/BeatmapAttributeText.cs

```diff
@@ -0,0 +1,133 @@
+        }.ToImmutableDictionary();
+
+        private readonly OsuSpriteText text;
+
+        public BeatmapAttributeText()
+        {
+            AutoSizeAxes = Axes.Both;
+
+            InternalChildren = new Drawable[]
+            {
+                text = new OsuSpriteText
+                {
+                    Anchor = Anchor.CentreLeft,
+                    Origin = Anchor.CentreLeft,
+                    Font = OsuFont.Default.With(size: 40)
+                }
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            Attribute.BindValueChanged(_ => updateLabel());
+            Template.BindValueChanged(_ => updateLabel());
+            beatmap.BindValueChanged(b =>
+            {
+                updateBeatmapContent(b.NewValue);
+                updateLabel();
+            }, true);
+        }
+
+        private void updateBeatmapContent(WorkingBeatmap workingBeatmap)
+        {
+            valueDictionary[BeatmapAttribute.Title] = workingBeatmap.BeatmapInfo.Metadata.Title;
+            valueDictionary[BeatmapAttribute.Artist] = workingBeatmap.BeatmapInfo.Metadata.Artist;
+            valueDictionary[BeatmapAttribute.DifficultyName] = workingBeatmap.BeatmapInfo.DifficultyName;
+            valueDictionary[BeatmapAttribute.Creator] = workingBeatmap.BeatmapInfo.Metadata.Author.Username;
+            valueDictionary[BeatmapAttribute.Length] = TimeSpan.FromMilliseconds(workingBeatmap.BeatmapInfo.Length).ToFormattedDuration();
+            valueDictionary[BeatmapAttribute.RankedStatus] = workingBeatmap.BeatmapInfo.Status.GetLocalisableDescription();
+            valueDictionary[BeatmapAttribute.BPM] = workingBeatmap.BeatmapInfo.BPM.ToString(@"F2");
+            valueDictionary[BeatmapAttribute.CircleSize] = ((double)workingBeatmap.BeatmapInfo.Difficulty.CircleSize).ToString(@"F2");
+            valueDictionary[BeatmapAttribute.HPDrain] = ((double)workingBeatmap.BeatmapInfo.Difficulty.DrainRate).ToString(@"F2");
+            valueDictionary[BeatmapAttribute.Accuracy] = ((double)workingBeatmap.BeatmapInfo.Difficulty.OverallDifficulty).ToString(@"F2");
+            valueDictionary[BeatmapAttribute.ApproachRate] = ((double)workingBeatmap.BeatmapInfo.Difficulty.ApproachRate).ToString(@"F2");
+            valueDictionary[BeatmapAttribute.StarRating] = workingBeatmap.BeatmapInfo.StarRating.ToString(@"F2");
+        }
+
+        private void updateLabel()
+        {
+            string newText = Template.Value.Replace("{Label}", label_dictionary[Attribute.Value].ToString())
+                                     .Replace("{Value}", valueDictionary[Attribute.Value].ToString());
+
+            foreach (var type in Enum.GetValues(typeof(BeatmapAttribute)).Cast<BeatmapAttribute>())
+            {
+                newText = newText.Replace("{" + type + "}", valueDictionary[type].ToString());
+            }
+
+            text.Text = newText;
```

Comment:
The rest of this class gives off the appearance that this should support localisation, but it actually doesn't, due to the `.ToString()` calls here that just end up using fallback English values. Was the intent here to support localisation fully?

If yes, a *rough* implementation of this method that works with localisation follows (although I don't claim it to be a particularly smart or good one) follows:

```diff
diff --git a/osu.Game/Skinning/Components/BeatmapAttributeText.cs b/osu.Game/Skinning/Components/BeatmapAttributeText.cs
index 572b4ba818..69136b3bb1 100644
--- a/osu.Game/Skinning/Components/BeatmapAttributeText.cs
+++ b/osu.Game/Skinning/Components/BeatmapAttributeText.cs
@@ -103,15 +103,28 @@ private void updateBeatmapContent(WorkingBeatmap workingBeatmap)
 
         private void updateLabel()
         {
-            string newText = Template.Value.Replace("{Label}", label_dictionary[Attribute.Value].ToString())
-                                     .Replace("{Value}", valueDictionary[Attribute.Value].ToString());
+            string numberedTemplate = Template.Value.Replace("{Label}", "{0}")
+                                              .Replace("{Value}", $"{{{1 + (int)Attribute.Value}}}");
 
             foreach (var type in Enum.GetValues(typeof(BeatmapAttribute)).Cast<BeatmapAttribute>())
             {
-                newText = newText.Replace("{" + type + "}", valueDictionary[type].ToString());
+                numberedTemplate = numberedTemplate.Replace("{" + type + "}", $"{{{1 + (int)type}}}");
             }
 
-            text.Text = newText;
+            var args = valueDictionary.OrderBy(pair => pair.Key)
+                                      .Select(pair => pair.Value)
+                                      .Prepend(label_dictionary[Attribute.Value])
+                                      .Cast<object?>()
+                                      .ToArray();
+
+            try
+            {
+                text.Text = LocalisableString.Format(numberedTemplate, args);
+            }
+            catch
+            {
+                text.Text = "Invalid format!";
+            }
         }
     }

 ```

The try-catch at the end is an escape hatch - because with this implementation user input is more or less directly translated to `string.Format` calls, trying to input an open brace would crash because `{` is not a valid format string spec for `string.Format` (it needs to be either terminated as a placeholder, or if a literal brace was meant, it needs to be doubled up as `{{` to be escaped).